pkg/controller/core.oam.dev/v1beta1/application/dispatcher.go

```diff
@@ -210,11 +210,11 @@ func (h *AppHandler) generateDispatcher(appRev *v1beta1.ApplicationRevision, rea
 	return manifestDispatchers, nil
 }
 
-func getTraitDispatchStage(client client.Client, traitType string, appRev *v1beta1.ApplicationRevision) (StageType, error) {
+func getTraitDispatchStage(client client.Client, traitType string, appRev *v1beta1.ApplicationRevision, fctx map[string]string) (StageType, error) {
 	trait, ok := appRev.Spec.TraitDefinitions[traitType]
 	if !ok {
 		trait = &v1beta1.TraitDefinition{}
-		err := oamutil.GetCapabilityDefinition(context.Background(), client, trait, traitType)
+		err := oamutil.GetCapabilityDefinition(context.Background(), client, trait, traitType, fctx)
```

Comment:
You can get app annotation from `appRev.metadata.annotations[autoupdate]`

Reply:
We don't need to pass fctx from all the way. For example, in this function, you can just pass `appRev.metadata.annotations`